project-fob/Views/Home/QRCode.cshtml

```diff
@@ -0,0 +1,25 @@
+﻿<head>
+    <title>QRCode</title>
+    <meta name="viewport" content="width=device-width,initial-scale=1,user-scalable=no" />
+    <script type="text/javascript" src="~/js/qrcode.js"></script>
+</head>
+<body>
+    <input id="text" type="hidden" value=@ViewBag.url style="width:80%" /><br />
+    <div id="qrcode" style="width:500px; height:500px; margin-top:15px;"></div>
+    <script type="text/javascript">
+        var qrcode = new QRCode(document.getElementById("qrcode"), {
+            width: 500,
+            height: 500
+        });
+
+        function makeCode() {
+            var elText = document.getElementById("text");
+            
+            qrcode.makeCode(elText.value);
+        }
+
+        makeCode();
+        
+
+    </script>
+</body>
```

Comment:
You're right about this being a bit of an odd way to do it, but when I tried it on my machine it works OK using the value directly. i.e.
```
<body>
    <div id="qrcode" style="width:500px; height:500px; margin-top:15px;"></div>
    <script type="text/javascript">
        var qrcode = new QRCode(document.getElementById("qrcode"), {
            width: 500,
            height: 500
        });
        
        qrcode.makeCode("@ViewBag.url");
    </script>
</body>
```

Reply:
Tried it and it does work.